scripts/import2mimir.sh

```
@@ -0,0 +1,356 @@
+        :) echo "Option -$OPTARG requires an argument." >&2; exit 1 ;;
+    esac
+done
+
+# Check that the variable $CONFIG_FILE is set and non-empty
+[[ -z "${CONFIG_FILE+xxx}" ]] &&
+{ echo -e "\e[91m config filename unset" >&2; echo "\e[0m" >&2; exit 1; }
+[[ -z "$CONFIG_FILE" && "${CONFIG_FILE+xxx}" = "xxx" ]] &&
+{ echo -e "\e[91m config filename set but empty" >&2; echo "\e[0m" >&2; exit 1; }
+
+# Source $CONFIG_FILE
+if [[ -f ${CONFIG_FILE} ]]; then
+  log_info "Reading ${CONFIG_FILE}"
+  source "${CONFIG_FILE}"
+elif [[ -f "${SCRIPT_DIR}/${CONFIG_FILE}" ]]; then
+  log_info "Reading ${SCRIPT_DIR}/${CONFIG_FILE}"
+  source "${SCRIPT_DIR}/${CONFIG_FILE}"
+else
+  log_error "Could not find ${CONFIG_FILE} in the current directory or in ${SCRIPT_DIR}"
+  exit 1
+fi
+
+check_arguments
+[[ $? != 0 ]] && { log_error "Invalid arguments. Aborting"; exit 1; }
+
+check_requirements
+[[ $? != 0 ]] && { log_error "Invalid requirements. Aborting"; exit 1; }
+
+check_environment
+[[ $? != 0 ]] && { log_error "Invalid environment. Aborting"; exit 1; }
+
+restart_docker_es 9200
+[[ $? != 0 ]] && { log_error "Could not restart the elastic search docker. Aborting"; exit 1; }
+
+# The order in which the import are done into mimir is important!
+# First we generate the admin regions with cosmogony
+# Second we import the addresses with bano
+
+download_osm
+[[ $? != 0 ]] && { log_error "Could not download osm. Aborting"; exit 1; }
+
+generate_cosmogony
+[[ $? != 0 ]] && { log_error "Could not generate cosmogony. Aborting"; exit 1; }
+
+import_cosmogony
+[[ $? != 0 ]] && { log_error "Could not import cosmogony into mimir. Aborting"; exit 1; }
+
+download_bano
+[[ $? != 0 ]] && { log_error "Could not download bano. Aborting"; exit 1; }
+
+import_bano
+[[ $? != 0 ]] && { log_error "Could not import bano into mimir. Aborting"; exit 1; }
+
+import_osm
+[[ $? != 0 ]] && { log_error "Could not import osm into mimir. Aborting"; exit 1; }
+
+download_ntfs
```

You might want to download all data before starting to import them, this way it will fail faster

---

:+1: